### PR TITLE
DROID-91: MeatNet connection management

### PR DESCRIPTION
**Pull request by miwright2**
- Create DataLinkArbitrator
- Heuristics to determine if a device should be connected to or disconnected from.
- Connection and Disconnection
- Read Device Information on Connection and publish.
- Flow for publishing node firmware details upon connection.
- MeatNet: automatic connection and reconnection to timers.
- Auto-Reconnect: !MeatNet with autoReconnect = true, automatic reconnect after framework call to connect.  Continue to reconnect until framework call disconnect.

### CODE REVIEW

File: combustion-android-ble/src/main/java/inc/combustion/framework/service/DeviceManager.kt [hunk 1]
```diff
@@ -497,6 +497,13 @@ class DeviceManager(
      */
     fun stopDfuMode() = service.stopDfuMode()
 
+    /**
+     * State flow containing the firmware details for all nodes on the network.
+     */
+    fun getNetworkFirmwareState(): StateFlow<FirmwareState>? {
```
Comment (miwright2) [inline combustion-android-ble/src/main/java/inc/combustion/framework/service/DeviceManager.kt:503]: @nick-sasquatch -- this adds the flow for firmware information for network to be used on the device list screen.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/DataLinkArbitrator.kt [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    }
+
+    fun shouldUpdateOnOutOfRange(device: ProbeBleDeviceBase): Boolean {
+        // TODO: Implement Multiplexing Business Logic
+        return debuggingWithStaticLink(device)
+    }
+
+    fun shouldHandleAdvertisingPacket(device: ProbeBleDeviceBase): Boolean {
+        /*
+        return true
```
Comment (jjohnstz) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/DataLinkArbitrator.kt:204]: Is this still needed?  Maybe add a TODO comment

**Review by jjohnstz — APPROVED** [review]
Looks great.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/BleDevice.kt [hunk 3]
```diff
@@ -152,6 +163,7 @@ internal open class BleDevice (
                     }
 
                     isConnected.set(connectionState == DeviceConnectionState.CONNECTED)
+                    isDisconnected.set(state is State.Disconnected)
```
Comment (nick-sasquatch) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/BleDevice.kt:166]: I don't see any usages where `isConnected` and `isDisconnected` checks could get interleaved, but if there really is some `Atomic`-ness necessary here, I'd make sure that there's not something that could sneak in between these two and end up with both being true.

File: combustion-android-ble/src/main/java/inc/combustion/framework/service/DeviceManager.kt (hunk 1 above)
Comment (nick-sasquatch) [reply combustion-android-ble/src/main/java/inc/combustion/framework/service/DeviceManager.kt:503]: Thanks for getting this in there!

**Review by nick-sasquatch — APPROVED** [review]
Thanks for getting the firmware update checks in! I left the rest of the review for Jesse with the exception of a minor comment about something I saw scrolling down to the DFU stuff.

**Pull request merged**